### ISSUE TITLE
Possible bug in torch.nn.init ?

### ISSUE THREAD

**Issue by aitsc** [pytorch/pytorch#69196]
I run this code first, and the accuracy of the direct test without training is 1%
...
m = nn.Conv1d(embed_dim, out_channels, kernel_size)
nn.init.trunc_normal_(m.weight, std=0.1, a=-0.2, b=0.2)
np.save('aa_w.npy', m.weight.detach().numpy())
m.weight = nn.parameter.Parameter(torch.tensor(np.load('aa_w.npy')))
...

I then changed to this code and ran it again, and the accuracy of the direct test without training was 6.6%
...
m = nn.Conv1d(embed_dim, out_channels, kernel_size)
m.weight = nn.parameter.Parameter(torch.tensor(np.load('aa_w.npy')))
...

I tried trunc_normal_ and kaiming_uniform_, both of them need to save the initialized parameters first, and then re-run the program to read the saved parameters before it works, why is that? I used constant_ to initialize it and it works fine.

Environment: python=3.9.7, torch==1.10+cu113, 2080ti on ubuntu 20.04
